exporter/syslogexporter/formatter.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package syslogexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/syslogexporter"
+
+import (
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+func createFormatter(protocol string) formatter {
+	if protocol == protocolRFC5424Str {
+		return newRFC5424Formatter()
+	}
+	return newRFC3164Formatter()
+}
+
+type formatter interface {
+	format(plog.LogRecord) string
```

Comment:
should the formatter interface have a validation function that returns an error in the following scenarios

- a required field is empty
- The value for a field is not in the right format. For example:
   ```
   The PRI part MUST have three, four, or five characters and will be
   bound with angle brackets as the first and last characters.  The PRI
   part starts with a leading "<" ('less-than' character, %d60),
   followed by a number, which is followed by a ">"
```

Reply:
> should the formatter interface have a validation function that returns an error in the following scenarios
> 
> * a required field is empty
> * The value for a field is not in the right format. For example:
>   ```
>   The PRI part MUST have three, four, or five characters and will be
>   bound with angle brackets as the first and last characters.  The PRI
>   part starts with a leading "<" ('less-than' character, %d60),
>   followed by a number, which is followed by a ">"
>   ```

`format` function will always add the PRI part e.g. [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27799/files#diff-fd44958375c0625e37038008e6afcd03ad386b6e60db4fb1d4d1fdccb1c4a27dR30)

for example [time.Format ](https://pkg.go.dev/time#Time.Format) doesn't return any error

Reply:
makes sense, the formatter ensures a valid format for PRI and other fields, I wonder what kind of validation logic exists in the syslog receiver. The exporter probably doesn't need to do any field validation (apart from populating default values).

Reply:
I initially implemented the `formatter`'s `format` function to return `(string, error)` but then it occurred that the implementation doesn't ever return an error. As there is no erroneous input, there's no need for a validation function.